# 4th and Punt: Forgone Conclusion or Missed Opportunity?
There is nothing as sure as an NFL head coach's decision when faced with a 4th down deep in your own territory - you punt. There is nothing to think about. Anyone who knows anything about football understands the risk of failing to get enough yardage on 4th down is usually not worth the reward. It's a tale as old as time. But is it still true? I'm going to take a look at this past 2018 NFL season to put this dogmatic theory to the test. 

### The Data
The data was gathered from the website __[NFLsavant.com](http://www.nflsavant.com/about.php)__ I chose to only use the data for the most current season to test the larger theory of "Punt on 4th and long" on a smaller sample.  

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

df = pd.read_csv('pbp-2018.csv')
df

GameId    GameDate  Quarter  Minute  Second OffenseTeam  \
0      2018090600  2018-09-06        2       6       9         PHI   
1      2018090600  2018-09-06        2       0       0         NaN   
2      2018090600  2018-09-06        3       0       0         NaN   
3      2018090600  2018-09-06        4       2      19         NaN   
4      2018090600  2018-09-06        4       1      18         NaN   
5      2018090901  2018-09-09        1       0       0         NaN   
6      2018090901  2018-09-09        2       1      58         NaN   
7      2018090902  2018-09-09        1       1      27         NaN   
8      2018090902  2018-09-09        1       0       0         NaN   
9      2018090902  2018-09-09        2       3       1         IND   
10     2018090902  2018-09-09        3       0       0         NaN   
11     2018090903  2018-09-09        1       0       0         NaN   
12     2018090903  2018-09-09        3       5      23         NaN   
13     2018090903  2018-09-09        3       0       0         NaN   
14     2018090903  2018-09-09        4       3      49         MIA   
15     2018090903  2018-09-09        4       0      10         MIA   
16     2018090904  2018-09-09        1       0       0         NaN   
17     2018090904  2018-09-09        2       0       6         NaN   
18     2018090905  2018-09-09        3       0       0         NaN   
19     2018090905  2018-09-09        4       2       0         NaN   
20     2018090906  2018-09-09        2       2       0         NaN   
21     2018090908  2018-09-09        1       0      25         NaN   
22     2018090908  2018-09-09        2       7       8          KC   
23     2018090908  2018-09-09        4       1      57         NaN   
24     2018090909  2018-09-09        3      14      26         ARI   
25     2018090910  2018-09-09        3       0       0         NaN   
26     2018090910  2018-09-09        4       1      27         NaN   
27     2018090911  2018-09-09        1       0       0         NaN   
28     2018090911  2018-09-09        2       0       0         NaN   
29     2018090911  2018-09-09        4       2       0         NaN   
...           ...         ...      ...     ...     ...         ...   
44986  2018123015  2018-12-30        2       5      44         ARI   
44987  2018123015  2018-12-30        2       2      41         ARI   
44988  2018123015  2018-12-30        2       2       0         ARI   
44989  2018123015  2018-12-30        2       1      18         ARI   
44990  2018123015  2018-12-30        2       0      21         ARI   
44991  2018123015  2018-12-30        2       0      15         ARI   
44992  2018123015  2018-12-30        2       0      11         ARI   
44993  2018123015  2018-12-30        3       9      11         ARI   
44994  2018123015  2018-12-30        3       8      29         ARI   
44995  2018123015  2018-12-30        3       7      38         ARI   
44996  2018123015  2018-12-30        3       5      19         SEA   
44997  2018123015  2018-12-30        3       4      40         SEA   
44998  2018123015  2018-12-30        3       4      34         SEA   
44999  2018123015  2018-12-30        3       0      11         ARI   
45000  2018123015  2018-12-30        4       8      58         SEA   
45001  2018123015  2018-12-30        4       8      17         SEA   
45002  2018123015  2018-12-30        4       7      38         SEA   
45003  2018123015  2018-12-30        4       7      32         SEA   
45004  2018123015  2018-12-30        4       6      59         SEA   
45005  2018123015  2018-12-30        4       6      25         SEA   
45006  2018123015  2018-12-30        4       5       6         ARI   
45007  2018123015  2018-12-30        4       4      24         ARI   
45008  2018123015  2018-12-30        4       3      44         ARI   
45009  2018123015  2018-12-30        4       3       3         ARI   
45010  2018123015  2018-12-30        4       2      27         ARI   
45011  2018123015  20

This is a detailed account of the season and includes the following statistics:

In [5]:
list(df)

['GameId',
 'GameDate',
 'Quarter',
 'Minute',
 'Second',
 'OffenseTeam',
 'DefenseTeam',
 'Down',
 'ToGo',
 'YardLine',
 'Unnamed: 10',
 'SeriesFirstDown',
 'Unnamed: 12',
 'NextScore',
 'Description',
 'TeamWin',
 'Unnamed: 16',
 'Unnamed: 17',
 'SeasonYear',
 'Yards',
 'Formation',
 'PlayType',
 'IsRush',
 'IsPass',
 'IsIncomplete',
 'IsTouchdown',
 'PassType',
 'IsSack',
 'IsChallenge',
 'IsChallengeReversed',
 'Challenger',
 'IsMeasurement',
 'IsInterception',
 'IsFumble',
 'IsPenalty',
 'IsTwoPointConversion',
 'IsTwoPointConversionSuccessful',
 'RushDirection',
 'YardLineFixed',
 'YardLineDirection',
 'IsPenaltyAccepted',
 'PenaltyTeam',
 'IsNoPlay',
 'PenaltyType',
 'PenaltyYards']

The total number of plays recorded in the 2018 season is:

In [2]:
print(len(df))

45016


Out of the 45016 plays I want to know how many were 4th down plays:

In [9]:
fourth = df[df['Down'] == 4]
len(fourth)

3764

In [ ]:
Or, about 